### PR TITLE
KT-19629 "Convert to primary constructor" quick fix should not move `init{...}` section down

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-19629

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/intentions/ConvertSecondaryConstructorToPrimaryIntention.kt [hunk 1]
```diff
@@ -181,10 +181,15 @@ class ConvertSecondaryConstructorToPrimaryIntention : SelfTargetingRangeIntentio
         with (constructorInClass.replace(constructor)) {
             constructorCommentSaver.restore(this)
         }
-        element.delete()
 
-        if ((initializer.body as? KtBlockExpression)?.statements?.isNotEmpty() ?: false) {
-            klass.addDeclaration(initializer)
+        if ((initializer.body as? KtBlockExpression)?.statements?.isNotEmpty() == true) {
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/intentions/ConvertSecondaryConstructorToPrimaryIntention.kt]: Why not simply add after `element` and then delete `element`?

Comment (t-kameyama) [reply idea/src/org/jetbrains/kotlin/idea/intentions/ConvertSecondaryConstructorToPrimaryIntention.kt]: I think property used in initializer needs to be defined before initializer.

**Comment by mglukhikh** [comment]
I think generally it's good now, but not exactly sure about `ReferencesSearch` usage (it can affect performance). I'll think a little bit and accept it.

**Comment by mglukhikh** [comment]
Finally I merged it. Thank you and sorry for so long reviewing!

**Pull request closed without merging**